Testscripts/Linux/Test_LIS_disk_check.sh

```diff
@@ -0,0 +1,193 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+# Description:
+# We've added new feature in LIS RPM installation script to check the disk space
+#       before proceeding to LIS install.
+# This avoids half / corrupted installations.
+# This test script executes in two ways -
+# Positive scenarios - Test leaves "bare minimum" size avaialble
+#       for LIS install and checks if LIS installation is successful.
+# Negative scenario - Test leaves "non installable" size on disk and
+#       checks if ./install.sh script skips the installation of not.
+# Note: This also takes care of additional space required for 7.3 and 7.4 distros.
+#	       For this test, we're expecting VHD with no LIS installed.
+#	       If LIS is installed, then this script will uninstall it.
+##################################################################################
+
+# Source utils.sh
+. utils.sh || {
+	echo "Error: unable to source utils.sh!"
+	echo "TestAborted" > state.txt
+	exit 0
+}
+
+# Source constants file and initialize most common variables
+UtilsInit
+
+PrepareForTest() {
+	lis_tarball_link=$1
+	if [ $(which rpm | wc -l) -gt 0 ]; then
+		total_hyperv_packages=$(rpm -qa | grep "hyper-v" | wc -l)
+		if [ $total_hyperv_packages -gt 0 ]; then
+			hyperv_package_names=$(rpm -qa | grep "hyper-v" | xargs)
+			LogMsg "Uninstalling hyperv packages..."
+			yum -y remove $hyperv_package_names
```

Comment:
confused with original design. I thought this is just checking disk size for LIS installation. why does it remove the package?

Reply:
@juhlee-microsoft This test simulates low disk space / bare minimum disk space scenario and attempts LIS installation. So, as prerequisite, we're removing the LIS (if preinstallied).